### ISSUE TITLE
inefficient handling of documents and coding process

### ISSUE THREAD

**Issue by hlageek** [RE-QDA/requal#34]
Currently, documents are render after each addition or removal of a code. Complex calculations are used to determine tag positions. For alpha release, we need to:
- [ ] separate writing to DB from document display
    - [ ] document display should be calculated only once, upon load, and more efficiently
    - [ ] coding actions should be fully handled by JavaScript

**Comment by hlageek** [comment]
I am closing this due to the backend changes implemented in https://github.com/RE-QDA/requal/pull/166. Although there is still room for improvement in efficiency, the current solutions seem to perform reasonably well. If the concerns about efficiency of the coding process reappear, either this issue can be reopened or we can start a new issue with more specificity.
